src/context/context.ts

```diff
@@ -38,6 +38,7 @@ export class Context {
     this.globals = renderOptions.globals ?? opts.globals
     this.environments = env
     this.strictVariables = renderOptions.strictVariables ?? this.opts.strictVariables
+    this.opts.ownPropertyOnly = renderOptions.ownPropertyOnly ?? opts.ownPropertyOnly
```

Comment:
I think we need a `this.ownPropertyOnly` instead of `this.opts`. The test case should be something like:

```typescript
const liquid = new Liquid({ ownPropertyOnly: false })

liquid.parseAndRender(tpl, {}, { ownPropertyOnly: true })
// expect using only ownproperty

liquid.parseAndRender(tpl, {})
// expect using ascendants' property
```